receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -206,6 +226,8 @@ func (kr *k8sobjectsreceiver) startPull(ctx context.Context, config *K8sObjectsC
 	kr.mu.Lock()
 	kr.stopperChanList = append(kr.stopperChanList, stopperChan)
 	kr.mu.Unlock()
+	kr.wg.Add(1)
```

Comment:
I wonder if `wg.Add(1)` should be moved under the lock 🤔 .

My fear is that if `stopWatches` is called in between, it may call `wg.Wait()` before the `wg.Add(1)`, which will probably panic? WDYT?

Reply:
Sounds good, thank you!